pkgs/tools/misc/uudeview/default.nix

```diff
@@ -1,18 +1,35 @@
-{ stdenv, fetchurl, tcl, tk }:
+{ stdenv
+, fetchurl
+, fetchpatch
+, tcl
+, tk
+}:
 
 stdenv.mkDerivation rec {
-  name = "uudeview-0.5.20";
+  pname = "uudeview";
+  version = "0.5.20";
+
   src = fetchurl {
-    url = "http://www.fpx.de/fp/Software/UUDeview/download/${name}.tar.gz";
+    url = "http://www.fpx.de/fp/Software/UUDeview/download/${pname}-${version}.tar.gz";
     sha256 = "0dg4v888fxhmf51vxq1z1gd57fslsidn15jf42pj4817vw6m36p4";
   };
 
   buildInputs = [ tcl tk ];
-  hardeningDisable = [ "format" ];
+
   configureFlags = [ "--enable-tk=${tk.dev}" "--enable-tcl=${tcl}" ];
 
-  # https://wiki.tcl.tk/3577
-  patches = [ ./matherr.patch ];
+  patches = [
+    # https://wiki.tcl.tk/3577
+    ./matherr.patch
+    # format hardening
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/OpenMandrivaAssociation/uudeview/master/uudeview-0.5.20-fix-str-fmt.patch";
+      sha256 = "1iyvpn3p0fmq3l1qidvgp7xfiwwj05649yycah4a7bynnz7y4xjh";
+    })
+  ];
+
+  patchFlags = [ "-p0" ]; # needed for fetched patch
```

Comment:
You don't need this flag or a change in the existing patch if you use `fetchpatch { ...; extraPrefix = ""; }` in the `fetchpatch` call (make sure to update the `sha256` when you do).

Reply:
> You don't need this flag or a change in the existing patch if you use `fetchpatch { ...; extraPrefix = ""; }` in the `fetchpatch` call (make sure to update the `sha256` when you do).

Great. Done.